Sources/Widgets/Manipulators/CPRManipulator/index.js

```diff
@@ -0,0 +1,126 @@
+
+export function intersectDisplayWithPlane(
+  x,
+  y,
+  planeOrigin,
+  planeNormal,
+  renderer,
+  glRenderWindow
+) {
+  const near = glRenderWindow.displayToWorld(x, y, 0, renderer);
+  const far = glRenderWindow.displayToWorld(x, y, 1, renderer);
+
+  return vtkPlane.intersectWithLine(near, far, planeOrigin, planeNormal).x;
+}
+
+// ----------------------------------------------------------------------------
+// vtkCPRManipulator methods
+// ----------------------------------------------------------------------------
+
+function vtkCPRManipulator(publicAPI, model) {
+  // Set our className
+  model.classHierarchy.push('vtkCPRManipulator');
+
+  publicAPI.handleEvent = (fromData, toData, glRenderWindow) => {
+    if (!model.stretchActor || !model.mapper) {
+      return {};
+    }
+
+    // Get normal and origin of the picking plane from the actor matrix
+    const stretchActorMatrix = [];
+    mat4.transpose(stretchActorMatrix, model.stretchActor.getMatrix());
+
+    const worldPlaneNormal = stretchActorMatrix.slice(8, 11); // 3rd column
+    const worldPlaneOrigin = stretchActorMatrix.slice(12, 15); // 4th column
+
+    // Convert world plane position to 2D position in the plane
+    const inverseStretchActorMatrix = [];
+    mat4.invert(inverseStretchActorMatrix, stretchActorMatrix);
+    const worldPlanePicking = intersectDisplayWithPlane(
+      toData.position.x,
+      toData.position.y,
+      worldPlaneOrigin,
+      worldPlaneNormal,
+      toData.pokedRenderer,
+      glRenderWindow
+    );
+    const modelPlanePicking = []; // (x, height - distance, 0)
+    vec3.transformMat4(
+      modelPlanePicking,
+      worldPlanePicking,
+      inverseStretchActorMatrix
+    );
+    const height = model.mapper.getHeight();
+    const distance = height - modelPlanePicking[1];
+
+    return publicAPI.distanceEvent(distance);
+  };
+
+  publicAPI.distanceEvent = (distance) => {
+    const height = model.mapper.getHeight();
+    const clampedDistance = Math.max(0, Math.min(height, distance));
+    const { position, orientation } =
+      model.mapper.getCenterlinePositionAndOrientation(clampedDistance);
+
+    let worldAxes;
+    if (orientation) {
+      const modelDirections = mat3.fromQuat([], orientation);
+      const baseDirections = model.mapper.getDirectionMatrix();
+      worldAxes = mat3.mul([], modelDirections, baseDirections);
+    }
+
+    model.lastDistance = clampedDistance;
+    return { worldPosition: position, worldAxes };
+  };
+
+  publicAPI.handleScroll = (nbSteps) => {
+    const distance = model.lastDistance + model.distanceStep * nbSteps;
+    return publicAPI.distanceEvent(distance);
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+function defaultValues(initialValues) {
+  return {
+    angle: 0,
+    distance: 0,
+    distanceStep: 1,
```

Comment:
I can't use max norm as the direction of projection depends on the point on the centerline
This would make scrolling speed dependent of point on the centerline

Reply:
I got it wrong, I meant `distanceStep= min(spacingInX, spacingInY, spacingInZ)`.
Maybe that could be a good default value or the value to consider if `distanceStep` is 0, null or undefined.

Reply:
Good idea!
This is actually what I use in the example: `cprManipulator.setDistanceStep(Math.min(...imageSpacing));`